### ISSUE TITLE
Allow wrapped block components of the same type to be adjacent

### ISSUE THREAD

**Issue by jkhaui** [facebook/draft-js#2162]
Hey,

My requirement is that I use Draft.js to build a paginated document editor (i.e. a "lite" google docs or MSWord). After experimenting, the best/only way seems to be using a custom "Page" wrapper component around a group of unstyled blocks. It works well so far... except for the fact that Draft automatically merges adjacent blocks of certain types:

![image](https://user-images.githubusercontent.com/36477474/63354276-8469ed00-c3a7-11e9-8568-be0385c8a022.png)

To be clear, the structure in that screenshot is:
- Styled-Component div (wrapping a group of 'unstyled' blocks)
- h3 block as a separator
- Styled-Component div

When I really need:
- Styled-Component div
- Styled-Component div

In other words, it's impossible for the editor to have 2 consecutive page nodes without an intermediary block (e.g. h2, h3, code-block, atomic, etc.) separating them. If the "separator" is not there, then the editor turns into a single never-ending page because the contents of the adjacent page nodes are merged together and re-wrapped within a single parent node.

I think this is the code responsible: https://github.com/facebook/draft-js/blob/7b9a7e116ab6dd639ebba28736629dfc04cbc7b0/src/component/contents/DraftEditorContents-core.react.js#L246
Or perhaps here: https://github.com/facebook/draft-js/blob/585af35c3a8c31fefb64bc884d4001faa96544d3/src/component/contents/exploration/DraftEditorBlockNode.react.js#L85
Is there any way I can overcome my issue through Draft's API or will I need to fork the project?

**Comment by jkhaui** [comment]
Closing this issue. I found a workaround; it's really hacky, but seems to be the only option given that Draft doesn't have a tree structure. I solved it by assigning 2 different block types the exact same elements and wrappers:


```
const blockRenderMap = Immutable.Map({
  'unstyled': {
    element: 'div',
    wrapper: <CustomPageBlock />
  },
  'code-block': {
    element: 'div',
    wrapper: <CustomPageBlock />
  }
});
```
Then when I programmatically create a new block, I alternate between 'unstyled' or 'code-block' depending on what the previous block was. This "tricks" Draft into recognising adjacent blocks as different types and therefore not group them, even though identical elements are rendered to the DOM.

It would be ideal if Draft eventually moves to a tree structure and has native support for nested blocks
